squigglepy/distributions.py

```diff
@@ -40,6 +41,14 @@ def __str__(self):
     def __repr__(self):
         return str(self)
 
+    def plot(self, num_samples=None, bins=None):
+        num_samples = 1000 if num_samples is None else num_samples
+        bins = 200 if bins is None else bins
```

Comment:
Why not use `def plot(self, num_samples=1000, bins=200)`?

Reply:
Hey!

Two reasons:

# 1)
TL;DR: What I wrote in the bottom of the PR description:
> Regarding `num_samples=None, bins=None`, I decided to leave flexibility for the implementation to decide on the num_samples if the user defines the bins, and vice versa, but this isn't a strong opinion.

Longer: I can imagine that the user might want to get more samples (sacrificing performance), and that we'll have some clever algorithm to also use more bins in that situation, maybe. Or maybe we'll have a clever implementation to use more bins if each bin has too many values, I don't know.
And we can't do any of that if we can't (in `plot()`'s implementation) distinguish between "the user explicitly asked for 200 bins" and "the user doesn't care how many bins to use"

# 2)
Another way I'm thinking about `plot()`'s signature here:

[seaborn.displot()](https://seaborn.pydata.org/generated/seaborn.displot.html#seaborn.displot) seem to be a best practice for creating an API similar to what I'm trying to make here (according to my DS friend), and they've put much more thought into this than I have, and they decided to make most of their params `None` by default.
<img width="766" alt="image" src="https://user-images.githubusercontent.com/11378537/229380998-89efb25e-18e4-4b60-9113-18c95a949c71.png">

I can also imagine `BaseDistribution.plot()` getting a similar API to seaborn, and I want to start with an API that will "fit in", if I can, though this is secondary.

---

I'm also ok with making the change as you suggested. This isn't a strong opinion, I'm just sharing my considerations as you asked.

Reply:
Ok that's sensible. Thanks!